docs/3.0/resources/upgrade-to-prefect-3.mdx

```diff
@@ -169,3 +169,66 @@ This change affects you if: You're using agents from an early version of Prefect
 </info>
 
 In Prefect 2, agents were deprecated in favor of next-generation workers. Workers are now standard in Prefect 3. For detailed information on upgrading from agents to workers, please refer to our [upgrade guide](https://docs-3.prefect.io/3.0/resources/upgrade-agents-to-workers).
+
+### Resolving common gotchas
+
+#### `AttributeError: 'coroutine' object has no attribute <some attribute>`
+
+If you do **not** `await` an asynchronous function or method, this error will be raised when you try to use the object.
+
+To fix it, `await` the asynchronous function or method.
+
+For example, `Block`'s `load` method is asynchronous in an async context:
+<CodeGroup>
+```python Wrong
+from prefect.blocks.system import Secret
+
+async def my_async_function():
+    my_secret = Secret.load("my-secret")
+    print(my_secret.get()) # AttributeError: 'coroutine' object has no attribute 'get'
+```
+
+```python Correct
+from prefect.blocks.system import Secret
+
+async def my_async_function():
+    my_secret = await Secret.load("my-secret")
+    print(my_secret.get()) # This will work
+```
+</CodeGroup>
```

Comment:
I think another example where it's not obviously an async function call could be helpful; for example, calling `state.result()` within a _synchronous_ state change hook attached to an _asynchronous_ task.  The best way to resolve this IMO is `state.result(_sync=True)`

Reply:
ooh yeah thats a good callout - if i use a codegroup to keep things vertically concise, you think 2 examples is reasonable for this gotcha? since i do see `Secret().get()` often

Reply:
I don't think it's a huge deal since this search should also be covered by our issues - go with your instinct and we can always add to it later if it comes up a few times.

Reply:
added in [a3d4d56](https://github.com/PrefectHQ/prefect/pull/15271/commits/a3d4d569f56f4a78e0f32b43971651ffe8212f2c)